packages/kbn-datemath/src/index.js

```diff
@@ -19,9 +19,20 @@
 
 import moment from 'moment';
 
-const units = ['y', 'M', 'w', 'd', 'h', 'm', 's', 'ms'];
-const unitsDesc = units;
-const unitsAsc = [...unitsDesc].reverse();
+const unitsMap = {
+  ms: { weight: 1, type: 'fixed', base: 1 },
+  s: { weight: 2, type: 'fixed', base: 1000 },
+  m: { weight: 3, type: 'mixed', base: 1000 * 60 },
+  h: { weight: 4, type: 'mixed', base: 1000 * 60 * 60 },
+  d: { weight: 5, type: 'mixed', base: 1000 * 60 * 60 * 24 },
+  w: { weight: 6, type: 'calendar' },
+  M: { weight: 7, type: 'calendar' },
+  // q: { weight: 8, type: 'calendar' }, // TODO: moment duration does not support quarter
```

Comment:
adding `q` to this list introduces all sorts of problems later down the stack due to moment duration's lack of support for this unit